frontend/src/components/Primitives/Dialogs/UserListDialog/UserListDialog.tsx

```diff
@@ -50,128 +61,102 @@ const ListMembersDialog = React.memo<ListMembersDialogProps>(
       return listToBeSorted;
     };
 
-    // Method to close dialog
     const handleClose = () => {
       setSearchMember('');
-      setUsersChecked(sortUserList());
+      setLocalUsersList(sortUserList);
       setIsOpen(false);
     };
 
-    const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-      setSearchMember(event.target.value);
-    };
-
-    const handleClearSearch = () => {
-      setSearchMember('');
-    };
-
     const handleChecked = (id: string) => {
-      const updateCheckedUsers = usersChecked.map((user) =>
+      const updateCheckedUsers = localUsersList.map((user) =>
         user._id === id ? { ...user, isChecked: !user.isChecked } : user,
       );
 
-      setUsersChecked(updateCheckedUsers);
+      setLocalUsersList(updateCheckedUsers);
     };
 
     const handleSelectAll = () => {
-      const updateCheckedUsers = usersChecked.map((user) =>
+      const updateCheckedUsers = localUsersList.map((user) =>
         user._id !== session?.user.id ? { ...user, isChecked: !isCheckAll } : user,
       );
 
       setIsCheckAll(!isCheckAll);
-      setUsersChecked(updateCheckedUsers);
+      setLocalUsersList(updateCheckedUsers);
     };
 
     const handleUpdateUsers = () => {
       setSearchMember('');
-      saveUsers(usersChecked);
+      confirmationHandler(localUsersList);
     };
 
-    const filteredList = useMemo(() => {
-      const searchString = searchMember.toLowerCase().trim();
-
-      return usersChecked.filter((user) => {
-        const fullName = `${user.firstName.toLowerCase()} ${user.lastName.toLowerCase()}`;
-        const email = user.email.toLowerCase();
-        return (
-          email.includes(searchString) || fullName.includes(searchString) || searchMember === ''
-        );
-      });
-    }, [searchMember, usersChecked]);
-
-    // Sets User List from State
     useEffect(() => {
-      if (usersList.length <= 0) return;
-
-      setUsersChecked(sortUserList());
+      if (usersList.length > 0) {
+        setLocalUsersList(sortUserList());
+      }
```

Comment:
```suggestion
      if (usersList.length <= 0) return;
      setLocalUsersList(sortUserList());
```

It's a good practice to immediately return if the condition to execute the logic is not met